native/core/src/parquet/read/values.rs

```diff
@@ -285,6 +261,59 @@ impl PlainDecoding for Int32DateType {
     }
 }
 
+impl PlainDecoding for Int32TimestampMicrosType {
+    #[inline]
+    fn decode(src: &mut PlainDecoderInner, dst: &mut ParquetMutableVector, num: usize) {
+        let src_data = &src.data;
+        let byte_width = src.desc.type_length() as usize;
+        let num_bytes = byte_width * num;
+
+        {
+            let mut offset = src.offset;
+            for _ in 0..num {
+                let v = &src_data[offset..offset + byte_width] as *const [u8] as *const u8
+                    as *const i32;
+
+                // TODO: optimize this further as checking value one by one is not very efficient
+                unsafe {
+                    if unlikely(v.read_unaligned() < JULIAN_GREGORIAN_SWITCH_OFF_DAY) {
```

Comment:
should we use unlikely as its rust nightly feature?

Reply:
We have our own `unlikely` https://github.com/apache/datafusion-comet/blob/main/native/core/src/lib.rs#L126